annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -507,6 +554,33 @@ private TranscriptConsequenceSummary getCanonicalTranscript(VariantAnnotation gn
         }
     }
 
+    private ColocatedVariant getGnomadData(VariantAnnotation gnResponse, MutationRecord record) {
+        if (gnResponse.getAnnotationSummary() != null &&
+            gnResponse.getColocatedVariants() != null &&
+            gnResponse.getColocatedVariants().size() > 0) {
+
+            ColocatedVariant variant = gnResponse.getColocatedVariants().get(0);
+            // get TUMOR_SEQ_ALLELE1 from mutation record
+            String tumorSeqAllele1 = record.getTUMOR_SEQ_ALLELE1();
+
+            // check if the dbSnpId start with "rs", now we have multiple ids returned in dbSnpId, such as "CM112509" or "COSM476".
+            // this checking can be removed after fixing the genome nexus model
+            if (variant.getDbSnpId().startsWith("rs")) {
+                // check if the gnomad_nfe_allele, gnomad_afr_allele and gnomad_eas_allele matches the tumorSeqAllele1
+                if (tumorSeqAllele1 != null && 
+                    (variant.getGnomadAfrAllele() != null && !tumorSeqAllele1.equals(variant.getGnomadAfrAllele()) ||
+                    variant.getGnomadEasAllele() != null && !tumorSeqAllele1.equals(variant.getGnomadEasAllele()) ||
+                    variant.getGnomadNfeAllele() != null && !tumorSeqAllele1.equals(variant.getGnomadNfeAllele()))) {
```

Comment:
@leexgh it would be good to have this logic in the web service (genome-nexus/genome-nexus) instead

Reply:
@inodb Sure will do it

Reply:
we decided to wait with moving this, b/c the format of colocatedvariants has changed for vep. Better to first update that then move this logic to the server